core/new-gui/src/app/workspace/service/workflow-graph/util/workflow-util.service.ts

```diff
@@ -60,4 +60,11 @@ export class WorkflowUtilService {
 
   }
 
+  public getNewOperatorLink(sourceOperator: OperatorPredicate, targetOperator: OperatorPredicate): OperatorLink {
+    const linkID = 'link-' + uuid();
+    const source = { operatorID: sourceOperator.operatorID, portID: sourceOperator.outputPorts[0]};
+    const target = { operatorID: targetOperator.operatorID, portID: targetOperator.inputPorts[0]};
+    return { linkID, source, target };
+  }
```

Comment:
it is not always true that its `outputPorts[0]` or `inputPorts[0]`, for instance, join operator and split operator.

Reply:
Move this implementation to drag-and-drop service and add another function `getRandomLinkID` that returns `'link-' + uuid()`. This function will be used in your logic to define a new operatorLInk.